test/jdk/java/lang/ProcessBuilder/ArgCheck.java

```diff
@@ -0,0 +1,314 @@
+            // The space in the argument requires it to be quoted, the final backslash
+            // must not be allowed to turn the quote that is added into a literal
+            // instead of closing the quote.
+            new CMD(true,
+                    ECHO_JAVA_ARGS,
+                    Arrays.asList(SPACE_AND_BACKSLASH, "ARG_21"),
+                    Arrays.asList(SPACE_AND_BACKSLASH, "ARG_21")),
+            new CMD(true,
+                    ECHO_CMD_ARGS,
+                    Arrays.asList(SPACE_AND_BACKSLASH, "ARG_22"),
+                    Arrays.asList(DOUBLE_QUOTE + SPACE_AND_BACKSLASH + BACKSLASH + DOUBLE_QUOTE, "ARG_22")),
+            new CMD(true,
+                    ECHO_VBS_ARGS,
+                    Arrays.asList(SPACE_AND_BACKSLASH, "ARG_23"),
+                    Arrays.asList(SPACE_AND_BACKSLASH + BACKSLASH, "ARG_23")),
+
+            // allowAmbiguousCommands = true, WITH application supplied double-quotes around the argument
+            // The argument has surrounding quotes so does not need further quoting.
+            // The backslash before the final quote is ignored and is interpreted differently for each command.
+            new CMD(true,
+                    ECHO_JAVA_ARGS,
+                    Arrays.asList(DOUBLE_QUOTE + SPACE_AND_BACKSLASH + DOUBLE_QUOTE, "ARG_31"),
+                    Arrays.asList("SPACE AND BACKSLASH\" ARG_31")),
+            new CMD(true,
+                    ECHO_CMD_ARGS,
+                    Arrays.asList(DOUBLE_QUOTE + SPACE_AND_BACKSLASH + DOUBLE_QUOTE, "ARG_32"),
+                    Arrays.asList(DOUBLE_QUOTE + SPACE_AND_BACKSLASH + DOUBLE_QUOTE, "ARG_32")),
+            new CMD(true,
+                    ECHO_VBS_ARGS,
+                    Arrays.asList(DOUBLE_QUOTE + SPACE_AND_BACKSLASH + DOUBLE_QUOTE, "ARG_33"),
+                    Arrays.asList(SPACE_AND_BACKSLASH, "ARG_33"))
+    );
+
+    /**
+     * Common function to Invoke a process with the commands and check the result.
+     *
+     * @param cmd a CMD test case with arguments, allowAmbiguousCommands mode, and expected output
+     */
+    private static void testCommand(CMD cmd) throws Exception {
+        System.setProperty(AMBIGUOUS_PROP_NAME, Boolean.toString(cmd.allowAmbiguous));
+        List<String> actual = null;
+        List<String> arguments = new ArrayList<>(cmd.command);
+        arguments.addAll(cmd.arguments);
+        try {
+            // Launch the process and wait for termination
+            ProcessBuilder pb = new ProcessBuilder(arguments);
+            Process process = pb.start();
+            try (InputStream is = process.getInputStream()) {
+                String str = readAllBytesAsString(is);
+                str = str.replace("\r", "");
+                actual = Arrays.asList(str.split("\n"));
+            } catch (IOException ioe) {
+                throw new RuntimeException(ioe.getMessage(), ioe);
```

Comment:
Do you think we should also explicitly do a `process.destroy` here, to force kill the launched process?